cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModelStats.java

```diff
@@ -257,42 +248,48 @@ public String toString() {
 
   /**
    * Generate statistics of utilization for resources among alive brokers in the given cluster.
+   * Average and standard deviation calculations are based on brokers not excluded for replica moves.
    *
    * @param clusterModel The state of the cluster.
+   * @param optimizationOptions Options to take into account while retrieving cluster capacity.
    */
-  private void utilizationForResources(ClusterModel clusterModel) {
+  private void utilizationForResources(ClusterModel clusterModel, OptimizationOptions optimizationOptions) {
     // Average, maximum, and standard deviation of utilization by resource.
-    Map<Resource, Double> avgUtilizationByResource = new HashMap<>();
-    Map<Resource, Double> maxUtilizationByResource = new HashMap<>();
-    Map<Resource, Double> minUtilizationByResource = new HashMap<>();
-    Map<Resource, Double> stDevUtilizationByResource = new HashMap<>();
+    Map<Resource, Double> avgUtilizationByResource = new HashMap<>(Resource.cachedValues().size());
+    Map<Resource, Double> maxUtilizationByResource = new HashMap<>(Resource.cachedValues().size());
+    Map<Resource, Double> minUtilizationByResource = new HashMap<>(Resource.cachedValues().size());
+    Map<Resource, Double> stDevUtilizationByResource = new HashMap<>(Resource.cachedValues().size());
     for (Resource resource : Resource.cachedValues()) {
-      double avgUtilizationPercentage = clusterModel.load().expectedUtilizationFor(resource) / clusterModel.capacityFor(resource);
+      double resourceUtilization = clusterModel.load().expectedUtilizationFor(resource);
+      double avgUtilizationPercentage = resourceUtilization / clusterModel.capacityWithAllowedReplicaMovesFor(resource, optimizationOptions);
       double balanceUpperThreshold = avgUtilizationPercentage * _balancingConstraint.resourceBalancePercentage(resource);
       double balanceLowerThreshold = avgUtilizationPercentage * Math.max(0, (2 - _balancingConstraint.resourceBalancePercentage(resource)));
       // Maximum, minimum, and standard deviation utilization for the resource.
       double hottestBrokerUtilization = 0.0;
       double coldestBrokerUtilization = Double.MAX_VALUE;
       double varianceSum = 0.0;
-      int numBalancedBrokers = 0;
+      int numBalancedBrokersInBrokersAllowedReplicaMove = 0;
```

Comment:
Nit: Maybe it's a personal preference. But I'd name this variable `numBalancedAllowedReplocaMoveBrokers`

Reply:
Since we are counting the number of balanced brokers in `_brokersAllowedReplicaMove`, I'd prefer to keep the current name.